### PR TITLE
[enhance](mtmv)support partition tvf

### PR DESCRIPTION
**Pull request by zddr**
The displayed content is consistent with the show partitions
 ```
mysql> select * from partitions("catalog"="internal","database"="zd","table"="ss")\G
*************************** 1. row ***************************
             PartitionId: 14004
           PartitionName: p1
          VisibleVersion: 1
      VisibleVersionTime: 2024-06-18 17:41:07
                   State: NORMAL
            PartitionKey: k3
                   Range: [types: [INT]; keys: [1]; ]
         DistributionKey: k2
                 Buckets: 2
          ReplicationNum: 1
           StorageMedium: HDD
            CooldownTime: 9999-12-31 23:59:59
     RemoteStoragePolicy: 
LastConsistencyCheckTime: \N
                DataSize: 0.000 
              IsInMemory: 0
       ReplicaAllocation: tag.location.default: 1
               IsMutable: 1
      SyncWithBaseTables: 1
            UnsyncTables: \N
1 row in set (0.02 sec)
```

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run performance

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/PartitionsTableValuedFunction.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+    public static final String NAME = "partitions";
+
+    private static final String CATALOG = "catalog";
+    private static final String DB = "database";
+    private static final String TABLE = "table";
+
+    private static final ImmutableSet<String> PROPERTIES_SET = ImmutableSet.of(CATALOG, DB, TABLE);
+
+    private static final ImmutableList<Column> SCHEMA = ImmutableList.of(
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/PartitionsTableValuedFunction.java:71]:
```suggestion
    private static final ImmutableList<Column> SCHEMA_FOR_OLAP_TABLE = ImmutableList.of(
```

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/PartitionsTableValuedFunction.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            new Column("RemoteStoragePolicy", ScalarType.createStringType()),
+            new Column("LastConsistencyCheckTime", ScalarType.createStringType()),
+            new Column("DataSize", ScalarType.createStringType()),
+            new Column("IsInMemory", ScalarType.createType(PrimitiveType.BOOLEAN)),
+            new Column("ReplicaAllocation", ScalarType.createStringType()),
+            new Column("IsMutable", ScalarType.createType(PrimitiveType.BOOLEAN)),
+            new Column("SyncWithBaseTables", ScalarType.createType(PrimitiveType.BOOLEAN)),
+            new Column("UnsyncTables", ScalarType.createStringType()));
+
+    private static final ImmutableList<Column> OTHER_SCHEMA = ImmutableList.of(
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/PartitionsTableValuedFunction.java:93]:
```suggestion
    private static final ImmutableList<Column> SCHEMA_FOR_EXTERNAL_TABLE = ImmutableList.of(
```

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Review by AshinGau — APPROVED** [review]

**Pull request merged**